include/sys/emmc.h

```diff
@@ -0,0 +1,279 @@
+#ifndef _SYS_EMMC_H_
+#define _SYS_EMMC_H_
+
+#include <inttypes.h>
+#include <sys/mimiker.h>
+#include <sys/device.h>
+#include <stddef.h>
+
+typedef enum emmc_op_mode {
+  EMMCMODE_BOOT,
+  EMMCMODE_CARD_ID,
+  EMMCMODE_INTR,
+  EMMCMODE_TRANSFER,
+  EMMCMODE_INA,
+} emmc_op_mode_t;
+
+typedef enum emmc_result {
+  EMMC_OK,
+  EMMC_ERR_IERR,
+  EMMC_ERR_TIMEOUT,
+  EMMC_ERR_NOT_RESPONDING,
+  EMMC_ERR_INTR,
+  EMMC_ERR_BUSY,
+} emmc_result_t;
```

Comment:
Note that this scheme of defining enum constants (not consecutive powers of 2) makes it harder to use binary operations like `&` and `|`.

Reply:
Those are meant to be mutually exclusive. No binary operation should be performed on these.